sdk/src/proto/network.rs

```diff
@@ -1,125 +1,234 @@
+#[derive(serde::Serialize, serde::Deserialize)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct GetProofStatusResponse {
+    /// The status of the proof request.
+    #[prost(enumeration = "ProofStatus", tag = "1")]
+    pub status: i32,
+    /// Optional proof url, where you can download the result of the proof request. Only included if
+    /// the proof has been fulfilled.
+    #[prost(string, tag = "2")]
+    pub proof_url: ::prost::alloc::string::String,
+}
+/// The request to get proof requests by a given status.
+#[derive(serde::Serialize, serde::Deserialize)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct GetProofRequestsRequest {
+    /// The status of the proof requests to get.
+    #[prost(enumeration = "ProofStatus", tag = "1")]
+    pub status: i32,
+}
+/// The response for getting proof requests by a given status.
+#[derive(serde::Serialize, serde::Deserialize)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct GetProofRequestsResponse {
+    /// The proof identifiers of the proof requests. Limited to the 10 most recent proof requests with
+    /// that status.
+    #[prost(string, repeated, tag = "1")]
+    pub proof_ids: ::prost::alloc::vec::Vec<::prost::alloc::string::String>,
+}
+/// The request to create a proof, the first step in requesting a proof.
 #[derive(serde::Serialize, serde::Deserialize)]
 #[allow(clippy::derive_partial_eq_without_eq)]
 #[derive(Clone, PartialEq, ::prost::Message)]
 pub struct CreateProofRequest {
+    /// The nonce for the account.
     #[prost(uint64, tag = "1")]
     pub nonce: u64,
+    /// The deadline for the proof request, signifying the latest time a fulfillment would be valid
+    /// for. Any time after this, provers will no longer continue to attempt to fulfill.
     #[prost(uint64, tag = "2")]
     pub deadline: u64,
+    /// The signature of the message.
     #[prost(bytes = "vec", tag = "3")]
     pub signature: ::prost::alloc::vec::Vec<u8>,
 }
+/// The response for creating a proof, the first step in requesting a proof.
 #[derive(serde::Serialize, serde::Deserialize)]
 #[allow(clippy::derive_partial_eq_without_eq)]
 #[derive(Clone, PartialEq, ::prost::Message)]
 pub struct CreateProofResponse {
+    /// The proof identifier.
     #[prost(string, tag = "1")]
     pub proof_id: ::prost::alloc::string::String,
+    /// The url to upload the elf to.
     #[prost(string, tag = "2")]
-    pub program_put_url: ::prost::alloc::string::String,
+    pub program_url: ::prost::alloc::string::String,
+    /// The url to upload the stdin to.
     #[prost(string, tag = "3")]
-    pub stdin_put_url: ::prost::alloc::string::String,
+    pub stdin_url: ::prost::alloc::string::String,
 }
+/// The request to submit a proof, the second step in requesting a proof. Should only be called after
```

Comment:
nit: Should only -> Returns an error if the program and stdin have not been uploaded yet.

Reply:
added this to client.rs